cmd/internal/cli/pull.go

```diff
@@ -228,13 +232,69 @@ func pullRun(cmd *cobra.Command, args []string) {
 			sylog.Fatalf("While creating Docker credentials: %v", err)
 		}
 
-		libexec.PullOciImage(name, args[i], types.Options{
-			TmpDir:           tmpDir,
-			Force:            force,
-			NoHTTPS:          noHTTPS,
-			DockerAuthConfig: authConf,
-			NoCleanUp:        noCleanUp,
-		})
+		sysCtx := &ocitypes.SystemContext{
+			OCIInsecureSkipTLSVerify:    noHTTPS,
+			DockerInsecureSkipTLSVerify: noHTTPS,
+			DockerAuthConfig:            authConf,
+		}
+
+		sum, err := ociclient.ImageSHA(args[i], sysCtx)
+		if err != nil {
+			sylog.Fatalf("Failed to get SHA of %v: %v", args[i], err)
+		}
+
+		name := uri.GetName(args[i])
+		imgabs := cache.OciTempImage(sum, name)
+
+		exists, err := cache.OciTempExists(sum, name)
+		if err != nil {
+			sylog.Fatalf("Unable to check if %v exists: %v", imgabs, err)
```

Comment:
Is it really ok to show this path to the user? Shouldn't you use the name provided by the user? Either `args[i]` or `name`?
```suggestion
			sylog.Fatalf("Unable to check if %s exists: %s", imgabs, err)
```